.github/CONTRIBUTING.md

```diff
@@ -50,192 +48,260 @@ npm run bootstrap
+  - Make a fresh clone of BotFramework-WebChat and run the test suites without any changes. If the fresh clone tests pass, this means the latest code changes are causing failures.
+- Identify whether the failure is intermittent
+  - Run the (failing) test suites again, potentially using Jest filters (<kbd>F</kbd> for failed tests)
 
-Our CI pipeline run tests with 4 agents simultaneously. If new tests are added, please make sure they can run simultaneously.
+If existing test suites fail without any code changes, please determine the following:
 
-### Troubleshooting the test suite
+- Test suites always fail, even after repeated runs
+  1.  The service may have been updated, causing the test suite failures. Please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) on the Web Chat repository.
+- Test suites fail intermittently
+  1.  Service reliability problems (e.g. for DirectLine, or Mockbot) may be the cause. Please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) on the Web Chat repository.
+  1.  Test reliability problems may be the cause. If so, please comment on [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938) and include:
+  - Failing test names/files
+  - Failing screencaps, if available. These images can be retrieved from `__diff_output__`
+  - Error messages
 
-We run test suite on every commit and requires 100% test pass. If the test suite did not complete successfully, they are likely:
+To debug race conditions:
 
--  Intermittent services issue
--  Test reliability issue, please see [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
--  Polluted development environment, for example:
-   -  Outdated `node_modules` content
-   -  Outdated Node.js or NPM
+1. Append `location.reload()` to the test code to continually reload the test page until it fails
+1. Navigate to the test page on `http://localhost:5001/tests/<testname>` and wait until the test fails, after which the automatic reloading will stop
 
-When the test suite failed:
+General tips on race conditions or intermittent test failures:
 
-1. Re-run the test suite
-   1. If it succeeded but a specific test fails intermittently, please comment to [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
-1. If it continues to fail, please fresh clone the repository and run the test suite without your changes
-   1. If you suspect your environment is polluted, please use [Windows Sandbox](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-sandbox/windows-sandbox-overview) and/or a new Linux distro on WSL to verify
-1. If it still fails repeatedly, please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) to us.
+- After sending a message, wait for responses from the bot (using `host.minMessagesShown(2)`)
+- After a long message is shown, wait for scroll to complete using `host.scrollComplete()`
```

Comment:
Sorry, the correct method name is `scrolToBottomCompleted` - could you confirm?

Reply:
Yap, fixed. `await pageConditions.scrollToBottomCompleted`.